documentation/src/main/asciidoc/reference/introduction.adoc

```diff
@@ -401,6 +401,41 @@ values!
 When you use Hibernate Reactive in Quarkus, these requirements are relaxed,
 and you can use public fields instead of getters and setters if you prefer.
 
+=== `equals()` and `hashCode()`
+
+Entity classes should override `equals()` and `hashCode()`. People new to
```

Comment:
> Also there's loads of literature on this topic, do we need to add more here?

I think we do. I don't see why somebody should look at this information somewhere else. I understand that other places could explain everything in deeper detail, but the documentation should at least give an idea about how we expect those method to be implemented for an entity.

Reply:
> Also I've never seen a solid and persuasive example in defence of the idea that entities really should declare hashcode and equals like you're stating here. 

I mean, I think it's just generally excepted that "data classes" should have "value equality", otherwise known as "equality".

Really "identity equality" is a sort-of garbage concept that doesn't really have a place outside of _very_ technical code, in my opinion.

(Try explaining the notion of "identity equality" to a functional programmer. It doesn't even _make sense_ in an environment which promotes immutability.)

Anyway, I would *always* define a proper notion of equality for all my entities. I need to anyway, because I certainly need to add the corresponding unique constraints to the database.

Reply:
You both have good points and I'm OK to soften my position, but please bear in mind that "identity equality" just works fine in all cases which don't involve re-attaching detached entities (which I dislike anyway) - and generally doesn't even need to be explained.

So for me it's still like the preferred approach and also simplest to implement.

This conversation really makes me think that after all these years we haven't solved this yet - perhaps we should take a step back and try to make this easier.

Reply:
Yes, sure, identity equality works, and I don't mind adding a clarification to state that.

But it seems to me that most users are going to want to define value equality, and it is very well-worth having clear guidance for those who do.

And y'know, I don't really trust the sources that they will find on google.

Reply:
Ok agreed. Thanks!

Reply:
I toned down the language a little, and added the proposed clarification.

I'm scared it's a bit too weak now, but whatever. Sure wouldn't want to hurt anyone's precious feelings.